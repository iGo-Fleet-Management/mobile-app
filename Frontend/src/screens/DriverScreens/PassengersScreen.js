import React, { useState } from 'react';
import { View, StyleSheet, ScrollView } from 'react-native';
import { SafeAreaView } from 'react-native-safe-area-context';

// Import components
import Header from '../../components/common/Header';
import SearchBar from '../../components/common/SearchBar';
import PassengerItem from '../../components/passengers/PassengerItem';
import AddPassengerButton from '../../components/passengers/AddPassengerButton';

const passageiros = [
  { id: 1, name: 'Hugo de Melo Carvalho', avatar: null },
  { id: 2, name: 'Iago Carro GuimarÃ£es', avatar: null },
  { id: 3, name: 'Lucas Barcelos Gomes', avatar: null },
  { id: 4, name: 'Paulo Henrique Reis', avatar: null },
  { id: 5, name: 'Rafael Galinari', avatar: null },
  { id: 6, name: 'Samuel Andrade', avatar: null },
];

export default function PassengersScreen({ navigation }) {
  const [search, setSearch] = useState('');

  const filteredPassageiros = passageiros.filter(passageiro => 
    passageiro.name.toLowerCase().includes(search.toLowerCase())
  );

  return (
    <SafeAreaView style={styles.container} edges={['top']}>
    <View style={styles.container}>
      <Header 
        title="Passageiros" 
      />

      <SearchBar 
        placeholder="Procurar passageiro"
        value={search}
        onChangeText={setSearch}
      />

      <ScrollView style={styles.listContainer}>
        {filteredPassageiros.map(passageiro => (
          <PassengerItem 
            key={passageiro.id}
            name={passageiro.name}
            avatar={passageiro.avatar}
            onPress={() => {/* Handle passenger selection */}}
          />
        ))}
      </ScrollView>

      <AddPassengerButton 
        onPress={() => navigation.navigate('AddPassenger')}
      />
    </View>
    </SafeAreaView>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#f5f5f5',
  },
  listContainer: {
    flex: 1,
  },
});