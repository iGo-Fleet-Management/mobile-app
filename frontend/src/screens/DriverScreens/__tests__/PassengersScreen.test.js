import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import PassengersScreen from '../PassengersScreen';

// Mock dos componentes
jest.mock('../../../components/common/Header', () => {
  const { Text, View } = require('react-native');
  return ({ title }) => (
    <View>
      <Text>{title}</Text>
    </View>
  );
});

jest.mock('../../../components/common/SearchBar', () => {
  const { TextInput, View } = require('react-native');
  return ({ placeholder, value, onChangeText }) => (
    <View>
      <TextInput 
        placeholder={placeholder}
        value={value}
        onChangeText={onChangeText}
      />
    </View>
  );
});

jest.mock('../../../components/passengers/PassengerItem', () => {
  const { TouchableOpacity, Text } = require('react-native');
  return ({ name, onPress }) => (
    <TouchableOpacity onPress={onPress}>
      <Text>{name}</Text>
    </TouchableOpacity>
  );
});

jest.mock('../../../components/passengers/AddPassengerButton', () => {
  const { TouchableOpacity, Text } = require('react-native');
  return ({ onPress }) => (
    <TouchableOpacity onPress={onPress}>
      <Text>Adicionar Passageiro</Text>
    </TouchableOpacity>
  );
});

jest.mock('react-native-safe-area-context', () => ({
  SafeAreaView: ({ children }) => children,
  useSafeAreaInsets: () => ({ top: 0, right: 0, bottom: 0, left: 0 })
}));

describe('PassengersScreen', () => {
  const mockNavigation = {
    navigate: jest.fn()
  };

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('deve renderizar o cabeçalho corretamente', () => {
    const { getByText } = render(
      <PassengersScreen navigation={mockNavigation} />
    );
    
    expect(getByText('Passageiros')).toBeTruthy();
  });

  it('deve renderizar a barra de pesquisa', () => {
    const { getByPlaceholderText } = render(
      <PassengersScreen navigation={mockNavigation} />
    );
    
    expect(getByPlaceholderText('Procurar passageiro')).toBeTruthy();
  });

  it('deve filtrar passageiros ao digitar na busca', () => {
    const { getByPlaceholderText, queryByText } = render(
      <PassengersScreen navigation={mockNavigation} />
    );
    
    const searchInput = getByPlaceholderText('Procurar passageiro');
    fireEvent.changeText(searchInput, 'Iago');
    
    expect(queryByText('Hugo de Melo Carvalho')).toBeNull();
    expect(queryByText('Iago Carro Guimarães')).toBeTruthy();
  });

  it('deve renderizar todos os passageiros inicialmente', () => {
    const { getByText } = render(
      <PassengersScreen navigation={mockNavigation} />
    );
    
    expect(getByText('Hugo de Melo Carvalho')).toBeTruthy();
    expect(getByText('Iago Carro Guimarães')).toBeTruthy();
    expect(getByText('Lucas Barcelos Gomes')).toBeTruthy();
    expect(getByText('Paulo Henrique Reis')).toBeTruthy();
    expect(getByText('Rafael Galinari')).toBeTruthy();
    expect(getByText('Samuel Andrade')).toBeTruthy();
  });

  it('deve navegar para adicionar passageiro ao pressionar o botão', () => {
    const { getByText } = render(
      <PassengersScreen navigation={mockNavigation} />
    );
    
    fireEvent.press(getByText('Adicionar Passageiro'));
    expect(mockNavigation.navigate).toHaveBeenCalledWith('AddPassenger');
  });

  it('deve renderizar o botão de adicionar passageiro', () => {
    const { getByText } = render(
      <PassengersScreen navigation={mockNavigation} />
    );
    
    expect(getByText('Adicionar Passageiro')).toBeTruthy();
  });

  it('deve corresponder ao snapshot', () => {
    const { toJSON } = render(
      <PassengersScreen navigation={mockNavigation} />
    );
    expect(toJSON()).toMatchSnapshot();
  });
});