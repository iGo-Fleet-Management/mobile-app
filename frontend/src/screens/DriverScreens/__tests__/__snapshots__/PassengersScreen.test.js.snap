// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PassengersScreen deve corresponder ao snapshot 1`] = `
<View
  style={
    {
      "backgroundColor": "#f5f5f5",
      "flex": 1,
    }
  }
>
  <View>
    <Text>
      Passageiros
    </Text>
  </View>
  <View>
    <TextInput
      onChangeText={[Function]}
      placeholder="Procurar passageiro"
      value=""
    />
  </View>
  <RCTScrollView
    style={
      {
        "flex": 1,
      }
    }
  >
    <View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      >
        <Text>
          Hugo de Melo Carvalho
        </Text>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      >
        <Text>
          Iago Carro Guimar√£es
        </Text>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      >
        <Text>
          Lucas Barcelos Gomes
        </Text>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      >
        <Text>
          Paulo Henrique Reis
        </Text>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      >
        <Text>
          Rafael Galinari
        </Text>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      >
        <Text>
          Samuel Andrade
        </Text>
      </View>
    </View>
  </RCTScrollView>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <Text>
      Adicionar Passageiro
    </Text>
  </View>
</View>
`;
